app/services/plotter.py

```
@@ -196,3 +199,77 @@ def plot_reorder_gap_hist(df: pd.DataFrame) -> Figure:
+        return fig
+    
+    # 차원 축소
+    if method == "tsne":
+        reducer = TSNE(n_components=2, random_state=42, perplexity=min(30, X.shape[0]-1))
+    else:
+        reducer = PCA(n_components=2, random_state=42)
+    
+    try:
+        X_2d = reducer.fit_transform(X)
+    except Exception as e:
+        print(f"차원 축소 실패: {e}", file=sys.stderr)
```

Ruff (0.12.2)

Undefined name `sys`

(F821)